cdap-ui/app/main.js

```diff
@@ -141,14 +141,14 @@ angular
       return {
         'request': function(config) {
           if ($rootScope.currentUser) {
-            angular.extend({
+            angular.extend(config, {
               user: $rootScope.currentUser || null,
               headers: {
                 // Accessing stuff from $rootScope is bad. This is done as to resolve circular dependency.
                 // $http <- myAuthPromise <- myAuth <- $http <- $templateFactory <- $view <- $state
-                authorization: ($rootScope.currentUser.token ? 'Bearer ' + $rootScope.currentUser.token: null)
+                Authorization: ($rootScope.currentUser.token ? 'Bearer ' + $rootScope.currentUser.token: null)
```

Comment:
:) yeah.. Angular has Dependency Injection and it will resolve for circular dependencies at runtime.